java/avro-converter/src/main/java/com/microsoft/AvroConverter.java

```diff
@@ -0,0 +1,133 @@
+package com.microsoft;
+
+import java.util.Map;
+import org.apache.avro.Schema.Parser;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.storage.Converter;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.experimental.models.MessageWithMetadata;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.serializer.TypeReference;
+import com.azure.data.schemaregistry.SchemaRegistryAsyncClient;
+import com.azure.data.schemaregistry.SchemaRegistryClientBuilder;
+import com.azure.data.schemaregistry.apacheavro.SchemaRegistryApacheAvroException;
+import com.azure.data.schemaregistry.apacheavro.SchemaRegistryApacheAvroSerializer;
+import com.azure.data.schemaregistry.apacheavro.SchemaRegistryApacheAvroSerializerBuilder;
+import com.azure.data.schemaregistry.models.SchemaRegistrySchema;
+import com.azure.identity.ClientSecretCredentialBuilder;
+import io.confluent.connect.avro.AvroData;
+import io.confluent.connect.avro.AvroDataConfig;
+import io.confluent.kafka.serializers.NonRecordContainer;
+
+public class AvroConverter implements Converter {
+  private SchemaRegistryAsyncClient schemaRegistryClient;
+  private SchemaRegistryApacheAvroSerializer serializer;
+  private SchemaRegistryApacheAvroSerializer deserializer;
+
+  private AvroData avroData;
```

Comment:
I don't recall seeing AvroData in our  KafkaAvroDeserializer and  KafkaAvroSerializer implementation (and they work with Avro generic/specific record as well), so what does this class get us that is not already provided in our serializer cases? It seems like either this is not needed here, or we are missing something in our serializers